app/src/main/java/com/feedhenry/securenativeandroidtemplate/AuthenticationFragment.java

```diff
@@ -5,200 +5,58 @@
- * A login screen that offers login via username/password.
+ * A login screen that offers login via Keycloak.
  */
 public class AuthenticationFragment extends Fragment {
 
-    View view;
-
-    private AuthenticateProvider mAuthProvider = new AuthenticateProviderImpl();
-    // Keep track of the login task to ensure we can cancel it if requested.
-    private UserLoginTask mAuthTask = null;
-
-    // UI references.
-    private AutoCompleteTextView mUsernameView;
-    private EditText mPasswordView;
-    private View mProgressView;
-    private View mLoginFormView;
-
-    private AuthenticateSuccessCallback mAuthSuccessCallback;
+    private View view;
 
     public AuthenticationFragment() {
         // Required empty public constructor
     }
 
-
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment - Set the view as the authenticate fragment
         view = inflater.inflate(R.layout.fragment_authentication, container, false);
 
-        // Reference the username and password fields from the UI
-        mUsernameView = (AutoCompleteTextView) view.findViewById(R.id.username);
-        mPasswordView = (EditText) view.findViewById(R.id.password);
-
-        // Add a listener for the keyboard input button so that the enter key will submit the form
-        mPasswordView.setOnEditorActionListener(new TextView.OnEditorActionListener() {
-            @Override
-            public boolean onEditorAction(TextView textView, int id, KeyEvent keyEvent) {
-                if (id == R.id.login || id == EditorInfo.IME_NULL) {
-                    attemptLogin();
-                    return true;
-                }
-                return false;
+        // Reference the keycloak login button
+        Button loginButton = (Button) view.findViewById(R.id.keycloakLogin);
+        loginButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                attemptKeycloakLogin();
             }
         });
 
-        // Reference the login button
-        Button mSignInButton = (Button) view.findViewById(R.id.sign_in_button);
-        mSignInButton.setOnClickListener(new OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                attemptLogin();
+        // Reference the keycloak logout button
+        Button logoutButton = (Button) view.findViewById(R.id.keycloakLogout);
+        logoutButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                attemptKeycloakLogout();
             }
         });
 
-        mLoginFormView = view.findViewById(R.id.login_form);
-        mProgressView = view.findViewById(R.id.login_progress);
+        // set the information dialog text
+        ((MainActivity)getActivity()).setInformationText(getString(R.string.popup_authentication_fragment));
```

Comment:
this could be called from the `MainActivity`, just before this fragment is being initialised. This way you won't have circular references between the `MainActivity` and the fragment.